src/base/Solution.cpp

```diff
@@ -8,14 +8,39 @@
 
 #include "cantera/base/Solution.h"
 #include "cantera/thermo/ThermoPhase.h"
+#include "cantera/thermo/ThermoFactory.h"
 #include "cantera/kinetics/Kinetics.h"
+#include "cantera/kinetics/KineticsFactory.h"
 #include "cantera/transport/TransportBase.h"
+#include "cantera/transport/TransportFactory.h"
 
 namespace Cantera
 {
 
 Solution::Solution() {}
 
+  Solution::Solution(const std::string& infile, std::string name, std::string transport,
+                     std::vector<shared_ptr<Solution> > adjacent) {
+
+    // thermo phase
+    m_thermo = shared_ptr<ThermoPhase>(newPhase(infile, name));
```

Comment:
The shorter way to write this is
```suggestion
    m_thermo.reset(newPhase(infile, name));
```